inspirehep/dojson/hep/fields/bd5xx.py

```diff
@@ -91,12 +91,32 @@ def hidden_note2marc(self, key, value):
 
 
 @hep.over('thesis', '^502..')
-@utils.filter_values
 def thesis(self, key, value):
     """Get Thesis Information."""
+    DEGREE_TYPES_MAP = {
+        'Diploma': 'Diploma',
+        'Bachelor': 'Bachelor',
+        'Laurea': 'Laurea',
+        'master': 'Master',
+        'Master': 'Master',
+        'thesis': 'Thesis',
+        'Thesis': 'Thesis',
+        'phD': 'PhD',
+        'Phd': 'PhD',
+        'PhD': 'PhD',
+        'PHD': 'PhD',
+        'PDF': 'PhD',
```

Comment:
What about upper-casing the input first, so that you automatically catch all case-related inconsistencies?